frontend/packages/console-app/src/components/nodes/NodeLogsUnitFilter.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from 'react';
+import { Chip, ChipGroup, TextInput } from '@patternfly/react-core';
+import { useTranslation } from 'react-i18next';
+import { getQueryArgument, setQueryArgument } from '@console/internal/components/utils';
+
+type NodeLogsUnitFilterProps = {
+  onChangeUnit: (value: string) => void;
+  unit: string;
+};
+
+const NodeLogsUnitFilter: React.FC<NodeLogsUnitFilterProps> = ({ onChangeUnit }) => {
+  const unitQueryArgument = 'unit';
+  const unitQueryParam = getQueryArgument(unitQueryArgument);
+
+  const [value, setValue] = React.useState('');
+  const [values, setValues] = React.useState(
+    unitQueryParam?.length > 0 ? unitQueryParam.split(',') : [],
+  );
+  const { t } = useTranslation();
+
+  const getValuesString = (valuesString: string[]) => valuesString.join(',');
+
+  React.useEffect(() => {
+    const listener = (event) => {
+      if ((event.code === 'Enter' || event.code === 'NumpadEnter') && value !== '') {
+        event.preventDefault();
+        values.push(value);
+        onChangeUnit(getValuesString(values));
+        setValue('');
+      }
+    };
+    document.addEventListener('keydown', listener);
+    return () => {
+      document.removeEventListener('keydown', listener);
+    };
+  }, [onChangeUnit, value, values]);
```

Comment:
Can you test if this works?

```suggestion
  }, []);
```

Otherwise you're going to be adding and removing the listener every time the value changes. I *think* that since the state variables are only used in the listener callback it's OK, but we'd need to check.

Reply:
The linter restores the values on save.  We have to bypass the linter...

Reply:
Yeah, you'd need to disable the linter for that line, which should be fine.

Reply:
```
  React.useEffect(() => {
    const listener = (event) => {
      if ((event.code === 'Enter' || event.code === 'NumpadEnter') && value !== '') {
        event.preventDefault();
        values.push(value);
        onChangeUnit(getValuesString(values));
        setValue('');
      }
    };
    document.addEventListener('keydown', listener);
    return () => {
      document.removeEventListener('keydown', listener);
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
```
breaks the filter (pressing enter no longer works).